haystack/components/joiners/document_joiner.py

```diff
@@ -78,22 +78,25 @@ def __init__(
         sort_by_score: bool = True,
     ):
         """
-        Create an DocumentJoiner component.
+        Creates a DocumentJoiner component.
 
         :param join_mode:
             Specifies the join mode to use. Available modes:
-            - `concatenate`
-            - `merge`
-            - `reciprocal_rank_fusion`
-            - `distribution_based_rank_fusion`
+            - `concatenate`: Keeps the highest-scored document in case of duplicates.
+            - `merge`: Calculates a weighted sum of scores for duplicates and merges them.
+            - `reciprocal_rank_fusion`: Merges and assigns scores based on reciprocal rank fusion.
+            - `distribution_based_rank_fusion`: Merges and assigns scores based on scores
+            distribution in each retriever.
         :param weights:
-            Weight for each list of Documents received, must have the same length as the number of inputs.
-            If `join_mode` is `concatenate` or `distribution_based_rank_fusion` this parameter is ignored.
+            Assign importance to each list of documents to influence how they're joined.
+            This parameter is used in join modes that combine scores and ignored for
```

Comment:
`distribution_based_rank_fusion` also kind of combines scores, so I'd suggest we can leave out this part

```suggestion
            This parameter is ignored for
```